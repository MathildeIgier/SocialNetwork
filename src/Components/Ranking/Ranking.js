import React from 'react';
import { View, Text, FlatList, Image } from 'react-native';

import styles from './RankingStyles';
import { imageStar } from '../../Assets/Images';

export default function Ranking({ rank }) {
  function getStars() {
    const stars = [];
    for (let i = 0; i < 5; i++) {
      const star = i >= rank ? imageStar.filled : imageStar.unfilled;
      stars.push(star);
    }
    return stars;
  }

  return (
    <View style={styles.starsContainer}>
      <FlatList
        horizontal={true}
        data={getStars()}
        renderItem={({ item }) => <Image style={styles.star} source={item} />}
        keyExtractor={(item, index) => index.toString()}
      />
    </View>
  );
}
