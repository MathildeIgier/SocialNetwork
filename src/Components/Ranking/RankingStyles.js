import { StyleSheet } from 'react-native';
import { Colors, ApplicationStyles } from '../../Themes';

export default StyleSheet.create({
  ...ApplicationStyles.screen,
  starsContainer: {
    paddingTop: 10,
    paddingBottom: 10
  },
  star: {
    height: 17,
    width: 17
  }
});
