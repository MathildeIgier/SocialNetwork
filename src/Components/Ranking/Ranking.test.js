// __tests__/Intro-test.js
import React from 'react';
import renderer from 'react-test-renderer';

import Ranking from './Ranking';

test('renders correctly', () => {
  const tree = renderer.create(<Ranking />).toJSON();
  expect(tree).toMatchSnapshot();
});
