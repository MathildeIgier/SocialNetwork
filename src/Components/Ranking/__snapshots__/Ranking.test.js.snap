// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "paddingBottom": 10,
      "paddingTop": 10,
    }
  }
>
  <RCTScrollView
    data={
      Array [
        Object {
          "testUri": "../../../src/Assets/Images/StarUnfilled.png",
        },
        Object {
          "testUri": "../../../src/Assets/Images/StarUnfilled.png",
        },
        Object {
          "testUri": "../../../src/Assets/Images/StarUnfilled.png",
        },
        Object {
          "testUri": "../../../src/Assets/Images/StarUnfilled.png",
        },
        Object {
          "testUri": "../../../src/Assets/Images/StarUnfilled.png",
        },
      ]
    }
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={true}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View>
      <View
        onLayout={[Function]}
        style={
          Array [
            Object {
              "flexDirection": "row",
            },
            null,
          ]
        }
      >
        <Image
          source={
            Object {
              "testUri": "../../../src/Assets/Images/StarUnfilled.png",
            }
          }
          style={
            Object {
              "height": 17,
              "width": 17,
            }
          }
        />
      </View>
      <View
        onLayout={[Function]}
        style={
          Array [
            Object {
              "flexDirection": "row",
            },
            null,
          ]
        }
      >
        <Image
          source={
            Object {
              "testUri": "../../../src/Assets/Images/StarUnfilled.png",
            }
          }
          style={
            Object {
              "height": 17,
              "width": 17,
            }
          }
        />
      </View>
      <View
        onLayout={[Function]}
        style={
          Array [
            Object {
              "flexDirection": "row",
            },
            null,
          ]
        }
      >
        <Image
          source={
            Object {
              "testUri": "../../../src/Assets/Images/StarUnfilled.png",
            }
          }
          style={
            Object {
              "height": 17,
              "width": 17,
            }
          }
        />
      </View>
      <View
        onLayout={[Function]}
        style={
          Array [
            Object {
              "flexDirection": "row",
            },
            null,
          ]
        }
      >
        <Image
          source={
            Object {
              "testUri": "../../../src/Assets/Images/StarUnfilled.png",
            }
          }
          style={
            Object {
              "height": 17,
              "width": 17,
            }
          }
        />
      </View>
      <View
        onLayout={[Function]}
        style={
          Array [
            Object {
              "flexDirection": "row",
            },
            null,
          ]
        }
      >
        <Image
          source={
            Object {
              "testUri": "../../../src/Assets/Images/StarUnfilled.png",
            }
          }
          style={
            Object {
              "height": 17,
              "width": 17,
            }
          }
        />
      </View>
    </View>
  </RCTScrollView>
</View>
`;
